src/Databases/DuckDB/fetchDuckDBTableStructure.cpp

```diff
@@ -0,0 +1,148 @@
+#include <Databases/DuckDB/fetchDuckDBTableStructure.h>
+
+#if USE_DUCKDB
+
+#include <Common/quoteString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDate32.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <Poco/String.h>
+
+#include <string_view>
+
+
+using namespace std::literals;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int DUCKDB_ENGINE_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+static DataTypePtr convertDuckDBDataType(String type)
+{
+    DataTypePtr res;
+    type = Poco::toLower(type);
+
+    if (type == "tinyint" || type == "int1")
+        res = std::make_shared<DataTypeInt8>();
+    else if (type == "utinyint")
+        res = std::make_shared<DataTypeUInt8>();
+    else if (type == "smallint" || type == "short" || type == "int2")
+        res = std::make_shared<DataTypeInt16>();
+    else if (type == "usmallint")
+        res = std::make_shared<DataTypeUInt16>();
+    else if (type == "integer" || type == "signed" || type == "int" || type == "int4")
+        res = std::make_shared<DataTypeInt32>();
+    else if (type == "uinteger")
+        res = std::make_shared<DataTypeUInt32>();
+    else if (type == "bigint" || type == "long" || type == "int8")
+        res = std::make_shared<DataTypeInt64>();
+    else if (type == "ubigint")
```

Comment:
`ubigint` looks ugly (like everything in DuckDB), but let's add it as a compatibility alias to ClickHouse (`registerDataTypeNumbers`), just in case, along with other type names.